### PR TITLE
Add timeline plot

### PR DESCRIPTION
**Pull request by eukaryo**
I have been implementing a new visualization function called "timeline plot". I want to test it with a visual regression test.
After merging this PR, the visual regression test should look like this.
https://github.com/eukaryo/optuna/pull/6
(The timeline plot with matplotlib is not merged yet. Some CIs are failed, but those are due to external factors.)

### CODE REVIEW

File: studies.py [hunk 1]
```diff
@@ -316,3 +317,47 @@ def objective(trial: optuna.Trial) -> float:
...
+        storage=storage,
+    )
+    study.enqueue_trial({"x": 0.9})
+    study.enqueue_trial({"x": 0.5})
+    study.enqueue_trial({"x": 0.3})
+    study.optimize(objective_timeline, n_trials=50, n_jobs=2, catch=(ValueError,))
+    studies.append((study.study_name, study))
+
+    # No trials single-objective study
+    optuna.create_study(study_name="A single objective study that has no trials", storage=storage)
```
Comment (toshihikoyanase) [inline studies.py:362]: This study was not added to `studies`. Do you intentionally exclude it?

**Review by toshihikoyanase — COMMENTED** [review]
I have a question on the usage of the empty study.

File: studies.py (hunk 1 above)
Comment (eukaryo) [reply studies.py:362]: No, it is a mistake. I have fixed the code to include the study on line#362.

**Comment by toshihikoyanase** [comment]
@eukaryo https://github.com/optuna/optuna-visualization-regression-tests/pull/9 was merged! Could you rebase the `main` branch to check the rendered results, please?

**Comment by toshihikoyanase** [comment]
Hmm, `plot_timeline` did not appear in the list. Do you have any idea about it?

image

https://6423c1930fe8584e4f44d3d0--effervescent-dieffenbachia-ab819a.netlify.app/

**Comment by toshihikoyanase** [comment]
The timeline plot was shown as expected when I executed the test locally.

```bash
$ python visual_regression_tests.py --heavy --output-dir ./public
```

image

image

**Comment by toshihikoyanase** [comment]
Ah, the netlify CI job checked out the current `main`, not feature branch.

c.f., https://github.com/optuna/optuna-visualization-regression-tests/commit/571e294d66b42859f81bbc7bc234cd6820d2e6d1

image

File: studies.py [hunk 2]
```diff
@@ -316,3 +316,50 @@ def objective(trial: optuna.Trial) -> float:
...
+
+    # Single-objective study
+    study = optuna.create_study(
+        study_name="A single objective study consuming time",
+        storage=storage,
+    )
+
+    study.enqueue_trial({"x": 0.3})
+    study.enqueue_trial({"x": 0.9})
+    study.enqueue_trial({"x": 0.5})
```
Comment (toshihikoyanase) [inline studies.py:342]: [nit] Could you add code comments to explain the expected trial states.
```suggestion
    study.enqueue_trial({"x": 0.3})  # Add a COMPLETE trial.
    study.enqueue_trial({"x": 0.9})  # Add a FAIL trial.
    study.enqueue_trial({"x": 0.5})  # Add a PRUNED trial.
```

File: studies.py [hunk 3]
```diff
@@ -316,3 +316,50 @@ def objective(trial: optuna.Trial) -> float:
...
+    study = optuna.create_study(
+        study_name=(
+            "A single objective study consuming time and "
+            "the order of legends is different from the order of trials"
+        ),
+        storage=storage,
+    )
+    study.enqueue_trial({"x": 0.9})
+    study.enqueue_trial({"x": 0.5})
+    study.enqueue_trial({"x": 0.3})
```
Comment (toshihikoyanase) [inline studies.py:356]: ditto.
```suggestion
    study.enqueue_trial({"x": 0.9})  # Add a FAIL trial.
    study.enqueue_trial({"x": 0.5})  # Add a PRUNED trial.
    study.enqueue_trial({"x": 0.3})  # Add a COMPLETE trial.
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. The plots are rendered as expected. I have some small comments, but I think they are optional.

We have some possible follow-up tasks as follows:
- Add RUNNING trials
- Add WAITING trials
- Add trials whose `datetime_complete` is earlier than `datetime_start`

The trials in the last item seem to be infeasible, but they actually exist because of the difference in the system clocks of computing nodes.

File: studies.py (hunk 1 above)
Comment (toshihikoyanase) [reply studies.py]: Thanks!

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Pull request merged**